### DISCUSSION TITLE
React-Tooltip V5

### DISCUSSION THREAD

**Discussion by danielbarion** [ReactTooltip/react-tooltip#812] [Polls]
I'm thinking about to re-write the entire tooltip from scratch to be easier maintenance and to receive PRs from the community.

But, I'm thinking about 2 ways and I want to know what everyone thinks will be the best approach:

1. Re-write the entire tooltip from scratch using react hooks and typescript (a very high effort, I don't know how much time that will take)
2. Re-write the tooltip using https://atomiks.github.io/tippyjs/v6/getting-started/ behind the scenes with typescript (if possible, I need to validate the idea too).
3. Re-write the tooltip using [floating-ui](https://floating-ui.com)https://floating-ui.com/ behind the scenes with typescript

thanks, @NemeZZiZZ for option 3!

If someone has other suggestions, please let us know!

I'll not edit the pool anymore to not lose the votes.

**Comment by NemeZZiZZ** [comment]
new option: https://floating-ui.com

**Comment by danielbarion** [comment]
Board for V5
https://github.com/ReactTooltip/react-tooltip/projects/2

**Comment by seiyab** [comment]
https://floating-ui.com/ itself seems good.
Then, why do we use react-tooltip rather than using floating-ui directly?

Using Tippy.js would make less improvement because I think many react-tooltip's problems come from manipulating DOM outside of React.

**Reply by danielbarion** [reply]
It's a good question @seiyab. Thinking about that, I believe using **floating-ui** behind the scenes will probably be less effort than creating everything from scratch, as far as I see, I'm the only active maintainer now in the project, the other guys help when they have time available, also, I don't have much time available and I don't want to let the package without maintenance.

So, I think if the code will be easier to update and fix, the community will help with the maintenance.

Does it make sense to you?

**Reply by seiyab** [reply]
Thank you for your response.

I probably understand.
My question compared no react-tooltip vs react-tooltip built with floating-ui.
However, react-tooltip already has so many users and it might grows since its name is likely to google.
Then, we should compare react-tooltip v4, that is very hard to maintain vs react-tooltip with floating-ui, that is very easy to maintain.
Then react-tooltip with floating-ui makes sense.

**Reply by danielbarion** [reply]
Yes, got your question now. Thanks for explaining!

**Comment by danielbarion** [comment]
If someone wants to check how the work is going, I'm updating almost every day the `V5` branch, so, you can just `git checkout V5`  and run `yarn` and `yarn dev`, after that, open `localhost:3000` in your browser and check :) 

There's a lot of work to be done until we get an alpha release ready to be tested by projects.

Also, we will probably have a lot of deprecations attributes, but I'm trying to build the tooltip version 5 from scratch with maximum compatibility, so, I'm trying to continue using the same prop names from <= v4.x.x;

Again, it is in his first step, so, I'm studying floating-ui API and building the tooltip at the same time, focusing on performance and functionality, I'll start working on TS types soon.

**Reply by danielbarion** [reply]
Also, some funding will be very welcome!
Thanks everyone :)

**Comment by danielbarion** [comment]
Closing due to V5, Option 3 selected.
